docs/typescript.md

```diff
@@ -47,48 +84,42 @@ const App = () => <Link href="#">Click me</Link>
 
 ### Passing Props
 
-You can type the props of your styled components.\
-Unfortunately, you will need to pass a second parameter with the tag name because TypeScript is unable to infer the tagname.
+You can type the props of your styled components.
 
 ```jsx
 import styled from 'react-emotion'
 
 type ImageProps = {
   src: string,
+  width: number;
 }
 
-const Image = styled<ImageProps, 'div'>('div')`
-  background: url(${props => props.src}) center center;
+const Image0 = styled('div')`
+  width: ${(props: ImageProps) => props.width};
```

Comment:
Is there a reason you are not using `styled<ImageProps, 'div'>('div')`? I think that looks way clearer, especially because then you don't have to do `(props: ImageProps) => ...` for every prop function.